src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesCheckPatterns.java

```diff
@@ -20,5 +20,19 @@ void method() {
 
         boolean result = (o instanceof String a) ?
                 (o instanceof String x) : (!(o instanceof String y));
+
+        boolean f =  (o instanceof String x);
+        // violation above, 'Unnecessary parentheses around assignment right-hand side.'
+
+        boolean b = (!(o instanceof Rectangle(_, _)));
+        // violation above, 'Unnecessary parentheses around assignment right-hand side.'
+        boolean c = !(o instanceof Rectangle(_, _));
```

Comment:
Given these new examples, shouldn't we have 2 violations on line 22 ?

Reply:
Yes, please open an issue for us to explore this more

Reply:
https://github.com/checkstyle/checkstyle/issues/15263